### ISSUE TITLE
Allow to run tests in verbose mode

### ISSUE THREAD

**Issue by nkaretnikov** [feuerbach/tasty#102]
It'd be nice to get verbose output via a command line flag or a special function. Something like what`verboseCheck` does, but for each tasty backend.

**Comment by feuerbach** [comment]
I don't know what verboseCheck does, can you give an example?

**Comment by nkaretnikov** [comment]
It shows the inputs and results, though it could be more informative (see below):

```
Prelude Test.QuickCheck> verboseCheck (\x -> x < 3)
Passed:
0
Passed:
1
Passed:
-1
Failed:
3
*** Failed! Passed:
0
Passed:
2
Falsifiable (after 4 tests): 
3
```

Currently, when a test fails, tasty shows the expected and actual values. I want to be able to see this information for all tests (regardless of whether a test succeeds or not). And I'd like this to work for all backends. It may look unnecessary for HUnit, for instance, but sometimes you need to write a complex test involving several function calls instead of, say, comparing string literals. In the former case, a verbose flag or function would allow me to see what my tests are actually doing. I could probably use `putStrLn` or `trace`, but it requires touching the code, which is error-prone.

**Comment by feuerbach** [comment]
I think of all existing providers QC is the only one that'd support such a mode. But that's fine, we can define the option itself in tasty-core, in case some other provider in the future will want to take advantage of it.

So you define the option somewhere in tasty-core (look at where the others are defined), then you handle it in quickcheck by running it in verbose mode and sticking the output into the result description.

Care to write a patch?

**Comment by nkaretnikov** [comment]
> Care to write a patch?

Okay, I'll try. I'll let you know regardless of the outcome.

**Comment by nkaretnikov** [comment]
[Here's what I've tried](https://github.com/nkaretnikov/tasty/commit/eacb458b7c88c78ab399507165779e6f8874fb36). The option shows up on `--help` (try tasty-quickcheck's testsuite), but it doesn't affect anything because `optionCLParser` always returns `Nothing`. Do you have time to look into it? If not, I'll try to continue next weekend.

**Comment by feuerbach** [comment]
I don't see anything obvious.

**Comment by nkaretnikov** [comment]
Just letting you know that I won't have much free time to spend on this in the near future. I hope I won't forget about it or just drop it in the longer term. That said, if someone else wanted to look into it in the meantime, that would be great.

**Comment by feuerbach** [comment]
Available in tasty-quickcheck-0.8.4 as `--quickcheck-verbose`
